pkg/controller/controller.go

```diff
@@ -522,3 +522,27 @@ func isInstanceReady(instance *v1alpha1.Instance) bool {
 
 	return false
 }
+
+func isOwnedByBinding(objectMeta *metav1.ObjectMeta, binding *v1alpha1.Binding) bool {
+	ref := getControllerRef(objectMeta)
+	if ref == nil {
+		return false
+	}
+	return ref.APIVersion == binding.APIVersion &&
+		ref.Kind == binding.Kind &&
+		ref.Name == binding.Name &&
```

Comment:
The only thing that should be checked about this reference is that the name of the referenced binding and the name of the passed in binding are the same.  APIVersion and Kind should not be checked.

Actually, it looks like you want to check both Name and UID.